qiskit/transpiler/preset_passmanagers/level2.py

```diff
@@ -329,7 +335,10 @@ def _contains_delay(property_set):
         pm2.append(_direction_check)
         pm2.append(_direction, condition=_direction_condition)
     pm2.append(_reset)
-    pm2.append(_depth_check + _opt + _unroll, do_while=_opt_control)
+
+    pm2.append(_depth_check + _opt + _unroll, do_while=_depth_control)
+    pm2.append(_size_check + _opt + _unroll, do_while=_size_control)
+
```

Comment:
This will always cause at least two additional runs through `_opt` and `_unroll`, even if `size` is already at a fixed point after the first one.  It would be better to condition on both the depth and the size on each loop iteration.

Reply:
- This change could easily be extended to the levels 1 and 3 although I'm not aware how much time overhead the passes would incur when searching for optimizing for size too. 

- I agree the test for this functionality may be fragile given further transpiler improvements. Other than that, could you elaborate upon what exactly you had in mind about the regression test? Thanks!

Reply:
A regression test in this case means using the example in the other instance, and writing assertions to verify that the circuit does not reduce its number of operations if you call transpile on the output of transpile.